apps/web/app/(app)/stats/LargestEmails.tsx

```diff
@@ -23,11 +23,15 @@ import { bytesToMegabytes } from "@/utils/size";
 import { formatShortDate } from "@/utils/date";
 import { Button } from "@/components/ui/button";
 import { getGmailUrl } from "@/utils/url";
-
+import { onTrashMessage } from "@/utils/actions-client";
+import { useState } from "react";
 export function LargestEmails(props: { refreshInterval: number }) {
   const session = useSession();
+  const [deletingStatus, setDeletingStatus] = useState<Record<string, boolean>>(
```

Comment:
You can change this to `isDeleting, setIsDeleting` and move it into the row level.

Reply:
Like this:

```tsx
const [isDeleting, setIsDeleting] = useState(false);
```

And then move it into the row (under `.map(() => {`)

Reply:
You can have it all local to the row. Don't think there's any need to pass that state up to the parent component.

Reply:
Create a new component to do useState